src/backend/routers/student.test.ts

```diff
@@ -81,6 +81,39 @@ test("addIep and getIep", async (t) => {
   t.deepEqual(got[0].end_date, added.end_date);
 });
 
+test("editIep", async (t) => {
+  const { trpc, seed } = await getTestServer(t, {
+    authenticateAs: "case_manager",
+  });
+
+  const start_date = new Date("2023-01-01");
+  const end_date = new Date("2023-12-31");
+
+  const added = await trpc.student.addIep.mutate({
+    student_id: seed.student.student_id,
+    start_date: start_date,
+    end_date: end_date,
+  });
+
+  const updated_start_date = new Date("2023-03-02");
+  const updated_end_date = new Date("2024-03-01");
+
+  const updated = await trpc.student.editIep.mutate({
+    student_id: seed.student.student_id,
+    start_date: updated_start_date,
+    end_date: updated_end_date,
+  });
+
+  const got = await trpc.student.getIeps.query({
+    student_id: seed.student.student_id,
+  });
+
+  t.is(got.length, 1);
+  t.is(got[0].student_id, seed.student.student_id);
+  t.deepEqual(got[0].start_date, updated.start_date);
```

Comment:
I would suggest doing a check for the start & end dates before the updated dates, and then showing the updated start & end dates are not the same as the previously added dates. You also don't need a const for `updated` since it shouldn't return anything. You should get your data with the `.getIeps.query()` instead.

Reply:
Just came here to say the same thing. I was playing around with this and tried to compare updated.start_date with updated_start_date and got the following difference showing in the test log:

  Difference (- actual, + expected):

  - Date 2023-03-01 08:00:00 UTC {}
  + Date 2023-03-02 00:00:00 UTC {}

Reply:
@hieungo89 , "updated" inserts the data into the iep keys, it doesn't retrieve anything. I can make "got" a let variable and reuse it after the data has been edited. re: your previous comment, if I make a test for checking the starting values, I will use "added" for that test. I can take out "updated" and simply have the await portion of the mutation of editIep, re-calling the "getIeps" with the "got" variable so I can use the new value in "notDeepEqual" tests.
Let me know if what I do is close to what you had in mind? I'll push it up once done with all comments.